### PR TITLE
replaced time.Sleep in main loop with ticker

### PR DESCRIPTION
**Pull request by gntouts**
There is no change in functionality, just replaced time.Sleep in main loop with ticker. 

Changes:

- Replace time.Sleep in main loop with ticker

Fixes sigstore/rekor-monitor#155

### CODE REVIEW

File: cmd/verifier/main.go [hunk 1]
```diff
@@ -101,7 +101,8 @@ func main() {
 	// 1. Fetch latest checkpoint and verify
 	// 2. If old checkpoint is present, verify consistency proof
 	// 3. Write latest checkpoint to file
-	for {
+	ticker := time.NewTicker(*interval)
```
Comment (haydentherapper) [inline cmd/verifier/main.go:104]: Can you add `defer ticket.Stop()` to clean up resources?

**Review by haydentherapper — CHANGES_REQUESTED** [review]
Thanks so much for the PR! Just one nit

File: cmd/verifier/main.go (hunk 1 above)
Comment (gntouts) [reply cmd/verifier/main.go:104]: Sure!

**Comment by haydentherapper** [comment]
The linter failed because unfortunately we currently have some bad design in the main loop where we call `log.Fatal` in a lot of places. We need to do some refactoring at some point, but for now, we can explicitly call `ticker.Stop()` before calling `log.Fatal`, or try to do some refactoring.

**Comment by gntouts** [comment]
Do you have any suggestions for refactoring? My initial thoughts would be to encapsulate a part of the main function within a separate function. This wrapper function should return an error containing the precise message that would be printed by log.Fatalf(). That way, the `defer ticker.Stop()` statement will run, while still allowing the use of log.Fatalf() in the main function to keep the user-facing messages unchanged.

**Comment by haydentherapper** [comment]
That sounds great. Over time, I'd like to split up the main loop into smaller functions, but for now, wrapping it would be good.

**Comment by gntouts** [comment]
I wrapped the loop in a separate function and modified the ticker loop to get an immediate fisrt tick. Let me know what you think.

**Pull request merged**